core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -546,6 +546,42 @@ class LogValidatorTest {
     checkOffsets(messageWithOffset, offset)
   }
 
+  @Test(expected = classOf[InvalidRecordException])
+  def testInvalidRecordCountTooManyNonCompressedV2() {
```

Comment:
There doesn't seem to be any particular need to put this here instead of DefaultRecordBatchTest. If we move them, then we can avoid the awkwardness of exposing a public static method on a test class.